### PR TITLE
Add cluster connection settings user input in cluster modal

### PR DESCRIPTION
**Pull request by arbulu89**
Implement the option to customize the ansible ssh connection settings, specifically the user that is used during the ssh connections.
This is done through the cluster settings modal.

It follows the next logic:
- User stored data goes to: `trento/v0/clusters/mycluster/user_settings/connection/` as map
- If the data is not stored, the code tries to use the Azure metadata information to get the OsProfile admin user name
- Ansible inventory is changed based on this logic, adding the `ansible_user` value

Demo:

connection_settings

Follow up issues:
https://github.com/trento-project/trento/issues/218
https://github.com/trento-project/trento/issues/219

### CODE REVIEW

**Comment by arbulu89** [comment]
@fabriziosestito @rtorrero @aleksei-burlakov @dottorblaster @stefanotorresi 
I need a review here to complete all the changes for the 0.4.0 release

File: web/clusters_test.go [hunk 1]
```diff
@@ -372,6 +382,19 @@ func checksResultByHost() map[string]*services.AggregatedCheckData {
...
+		"provider": cloud.Azure,
+		"metadata": &cloud.AzureMetadata{
+			Compute: cloud.Compute{
+				OsProfile: cloud.OsProfile{
+					AdminUserName: fmt.Sprintf("defuser%d", userIndex),
+				},
+			},
+		},
+	}
+}
```
Comment (dottorblaster) [inline web/clusters_test.go:396]: Very declarative, I like it!

**Review by dottorblaster — APPROVED** [review]

**Review by fabriziosestito — COMMENTED** [review]
@arbulu89  I tried to run the PR on gollum, the default user is an empty column in the modal.
Is this desired or we want to show "root" in the case we are not running on Azure?

image

P.S: Please don't laugh at my cluster petname

**Comment by arbulu89** [comment]
> @arbulu89 I tried to run the PR on gollum, the default user is an empty column in the modal.
> Is this desired or we want to show "root" in the case we are not running on Azure?
> 
> image
> 
> P.S: Please don't laugh at my cluster petname

This value is coming from the cloud metadata, so in libvirt we don't have it. Behind the scenes, if it is not being shown there `root` will be used.
The thing is that there is not any way of getting this default value, so using `root` would be a hardcoded value on the code.
We could put it there to have at least some value in the web ui

**Comment by fabriziosestito** [comment]
> > @arbulu89 I tried to run the PR on gollum, the default user is an empty column in the modal.
> > Is this desired or we want to show "root" in the case we are not running on Azure?
> > image
> > P.S: Please don't laugh at my cluster petname
> 
> This value is coming from the cloud metadata, so in libvirt we don't have it. Behind the scenes, if it is not being shown there `root` will be used.
> The thing is that there is not any way of getting this default value, so using `root` would be a hardcoded value on the code.
> We could put it there to have at least some value in the web ui

Oh ok, yes I would go with the hardcoded one in case we are not on azure, just to be consistent with the UI

**Comment by arbulu89** [comment]
> > > @arbulu89 I tried to run the PR on gollum, the default user is an empty column in the modal.
> > > Is this desired or we want to show "root" in the case we are not running on Azure?
> > > image
> > > P.S: Please don't laugh at my cluster petname
> > 
> > 
> > This value is coming from the cloud metadata, so in libvirt we don't have it. Behind the scenes, if it is not being shown there `root` will be used.
> > The thing is that there is not any way of getting this default value, so using `root` would be a hardcoded value on the code.
> > We could put it there to have at least some value in the web ui
> 
> Oh ok, yes I would go with the hardcoded one in case we are not on azure, just to be consistent with the UI

"root" default user now is showed in the modal if there is not any cloud metadata found:
image

**Review by fabriziosestito — APPROVED** [review]

**Pull request merged**